packages/next/next-server/server/config.ts

```diff
@@ -162,6 +162,15 @@ export default function loadConfig(
     }
 
     return assignDefaults({ configOrigin: CONFIG_FILE, ...userConfig })
+  } else {
+    const tsPath = findUp.sync(TS_CONFIG_FILE, {
```

Comment:
This should use the `path` std lib to strip the extension and test against numerous other extensions IMO. We may as well check for `/(tsx?|json)$/` and error for all three cases.